diff -Naurp kodi/xbmc/cores/VideoPlayer/DVDMessageQueue.h kodi-patch/xbmc/cores/VideoPlayer/DVDMessageQueue.h
--- kodi/xbmc/cores/VideoPlayer/DVDMessageQueue.h	2026-06-20 19:46:12.000000000 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/DVDMessageQueue.h	2026-06-21 10:49:42.000000000 +0000
@@ -80,7 +80,7 @@ public:
   void WaitUntilEmpty();
 
   // non messagequeue related functions
-  bool IsFull() const { return GetLevel(true) == 100; }
+  bool IsFull() const { return GetLevel() == 100; }
 
   /*!
    * \brief Get the current queue level.

