diff -Naurp kodi/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp kodi-patch/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp
--- kodi/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp	2026-06-24 15:57:05.955359026 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp	2026-06-24 15:59:28.000000000 +0000
@@ -751,7 +751,7 @@ void CRenderManager::Render(bool clear,
       PresentSingle(clear, flags, alpha);
   }
 
-  if (gui || m_renderDebug)
+  if (gui)
   {
     if (!m_pRenderer->IsGuiLayer())
       m_pRenderer->Update();
